Sources/ProtobufferCoding/TypeInfo.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+@_implementationOnly import Runtime
+
+// swiftlint:disable discouraged_optional_boolean
+
+// MARK: - Supported data types
+/// Supported primitive types.
+private let primitiveSupportedTypes: [Any.Type] = [
+    String.self,
+    Bool.self,
+    Int.self,
+    Int32.self,
+    Int64.self,
+    UInt.self,
+    UInt32.self,
+    UInt64.self,
+    Double.self,
+    Float.self,
+    Data.self,
+    String?.self,
+    Bool?.self,
+    Int?.self,
+    Int32?.self,
+    Int64?.self,
+    UInt?.self,
+    UInt32?.self,
+    UInt64?.self,
+    Double?.self,
+    Float?.self,
+    Data?.self
+]
+
+/// Supported arrays of primitive types.
+private let primitiveSupportedArrayTypes: [Any.Type] = [
+    [String].self,
+    [Bool].self,
+    [Int].self,
+    [Int32].self,
+    [Int64].self,
+    [UInt].self,
+    [UInt32].self,
+    [UInt64].self,
+    [Double].self,
+    [Float].self,
+    [Data].self,
+    [String?].self,
+    [Bool?].self,
+    [Int?].self,
+    [Int32?].self,
+    [Int64?].self,
+    [UInt?].self,
+    [UInt32?].self,
+    [UInt64?].self,
+    [Double?].self,
+    [Float?].self,
+    [Data?].self
+]
+
+internal func isPrimitiveSupported(_ type: Any.Type) -> Bool {
+    primitiveSupportedTypes.contains(where: { $0 == type })
```

Comment:
This looks very much like `Apodini/Utilities/TypeInfo.swift`. Sadly, an import would create cyclic dependencies 😕

Reply:
Yes, you're right. Parts of the code is even simply copied from `Apodini/Utilities/TypeInfo.swift`. But as you said, using `TypeInfo` from Apodini target does not work.